src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -656,11 +658,9 @@ internal override bool Equals(TypeSymbol t2, bool ignoreCustomModifiersAndArrayS
                 // If ignoring dynamic, compare underlying tuple types
                 if (t2.IsTupleType)
                 {
-                    t2 = ((TupleTypeSymbol)t2).UnderlyingTupleType;
-                    if ((object)t2 == this) return true;
+                    t2 = t2.TupleUnderlyingType;
+                    if (this.Equals(t2, ignoreCustomModifiersAndArraySizesAndLowerBounds, ignoreDynamic)) return true;
```

Comment:
That is a question for you :-), that is how you coded this initially. The change to symbol model shouldn't affect the logic.

---
In reply to: [61807976](https://github.com/dotnet/roslyn/pull/11006#discussion_r61807976) [](ancestors = 61807976)